control-plane/agents/core/src/core/specs.rs

```diff
@@ -141,20 +146,65 @@ pub trait SpecOperations: Clone + Debug + Sized + StorableObject + OperationSequ
                 }
             }
             Err(error) => {
-                let mut spec_clone = locked_spec.lock().clone();
-                spec_clone.clear_op();
-                let stored = registry.store_obj(&spec_clone).await;
+                // The create failed so delete the spec.
+                Self::delete_spec(registry, locked_spec).await?;
```

Comment:
I think we want to return the original error rather than the etcd error?
```suggestion
                Self::delete_spec(registry, locked_spec).await.ok();
```

Reply:
Done.